src/app/data-access/group-navigation.service.ts

```diff
@@ -1,28 +1,46 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-import * as D from 'io-ts/Decoder';
 import { appConfig } from 'src/app/utils/config';
-import { decodeSnakeCase } from 'src/app/utils/operators/decode';
-
-const groupNavigationChildDecoder = D.struct({
-  id: D.string,
-  name: D.string,
-  type: D.literal('Class', 'Team', 'Club', 'Friends', 'Other', 'User', 'Session', 'Base'),
-  currentUserManagership: D.literal('none', 'direct', 'ancestor', 'descendant'),
-  currentUserMembership: D.literal('none', 'direct', 'descendant'),
+import { decodeSnakeCaseZod } from 'src/app/utils/operators/decode';
+import { z } from 'zod';
+
+const groupNavigationChildSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  type: z.union([
```

Comment:
this is done everywhere using `z.enum`, see https://github.com/France-ioi/AlgoreaFrontend/blob/master/src/app/items/models/item-properties.ts